Themis/Views/Custom/LoadingSpinnerView.swift

```diff
@@ -0,0 +1,50 @@
+//
+//  LoadingSpinnerView.swift
+//  Themis
+//
+//  Created by Andreas Cselovszky on 22.12.22.
+//
+
+import SwiftUI
+
+struct LoadingSpinnerView: View {
+    var duration: Double = 0.7 // per spin
+    var lineWidth: Double = 8
+    @State var currDegree = Angle.degrees(540)
+
+    var body: some View {
+        ZStack {
+            Circle()
+                .stroke(
+                    Color.secondary.opacity(0.5),
+                    lineWidth: lineWidth
+                )
+            Circle()
+                .trim(from: 0, to: 0.25)
+                .stroke(
+                    Color.secondary,
+                    style: StrokeStyle(
+                        lineWidth: lineWidth,
+                        lineCap: .round
+                    )
+                )
+                .rotationEffect(currDegree)
+        }
+        .frame(width: 20)
+        .onAppear {
+            DispatchQueue.main.async {
+                withAnimation(.linear(duration: duration)
+                    .repeatForever(autoreverses: false)
+                ) {
+                    currDegree += .degrees(360)
+                }
+            }
+        }
+    }
```

Comment:
Why reinvent the wheel and use non-apple components? The following works.
(I had to search a white to find that I had to set the color manually, but otherwise I think this is the nicer, "more native" solution - although your custom spinner is also quite cool)

![screenshot-2023-01-02_000834](https://user-images.githubusercontent.com/9006596/210281314-f6f68079-7af0-45b1-86cf-60856897051c.gif)

```suggestion
    var body: some View {
        ProgressView()
            .frame(width: 20)
            .progressViewStyle(CircularProgressViewStyle(tint: Color.white))
    }
```

Reply:
I think it looks nicer as it also matches the score progress view. However this is just a matter of taste. I don't really care what we use in the end, as long as everybody likes it.